### PR TITLE
read generator config from mitosis.config.js

### LINKED ISSUE #199: Support reading generator options from `mitosis.config.js` in `mitosis compile`
Similar to how `mitosis build` works: it imports `getMitosisConfig()` and then override defaults:

https://github.com/BuilderIO/mitosis/blob/76bcc9732e213b7e736ee05117e66b5c40c8e787/packages/cli/src/commands/build.ts#L9
https://github.com/BuilderIO/mitosis/blob/76bcc9732e213b7e736ee05117e66b5c40c8e787/packages/cli/src/build/build.ts#L28-L55

Place to apply changes for `mitosis compile`: 

https://github.com/BuilderIO/mitosis/blob/76bcc9732e213b7e736ee05117e66b5c40c8e787/packages/cli/src/commands/compile.ts#L24-L27

### PR DESCRIPTION
**Pull request by sahilmob**
fixes #199

### CODE REVIEW

File: packages/cli/mitosis.config.js [hunk 1]
```diff
@@ -0,0 +1 @@
+module.exports = {};
```
Comment (samijaber) [inline packages/cli/mitosis.config.js:1]: Why is this file needed? 🤔

Comment (sahilmob) [reply packages/cli/mitosis.config.js:1]: Without this file tests won't pass because `require(process.cwd() + '/mitosis.config')` throws if `mitosis.config` doesn't exist.

The other solution is to wrap all the logic inside `getMitosisConfig` with a try catch: 

```js
import { MitosisConfig } from '@builder.io/mitosis';

export function getMitosisConfig(): MitosisConfig | null {
  try {
    const module = require(process.cwd() + '/mitosis.config');
    return module?.default || module || null;
  } catch (_) {
    return null
  }
}
```
What do you think?

Comment (samijaber) [reply packages/cli/mitosis.config.js:1]: Yeah, I think we should change the logic in `getMitosisConfig()` to check for the existence of the file before importing it. You can use `fs.existsSync(fileName)` for that

File: packages/cli/src/commands/compile.ts [hunk 2]
```diff
@@ -45,17 +46,19 @@ const command: GluegunCommand = {
       plugins.push(compileAwayBuilderComponents());
     }
 
+    const mitosisConfig = getMitosisConfig();
+    const generatorOptions = mitosisConfig?.options?.[to];
+
     const generatorOpts: Partial<{ [K in AllGeneratorOptionKeys]: any }> = {
+      ...generatorOptions,
       prettier: opts.prettier ?? true,
       plugins: plugins,
```
Comment (samijaber) [inline packages/cli/src/commands/compile.ts:55]: One small detail I noticed here: we have a `plugins` array that is declared here in the function. We want to merge it with the contents of `generatorOptions.plugins` if there are any. I think that'll look like this:

```suggestion
    const mitosisConfig = getMitosisConfig();
    const generatorOptions = mitosisConfig?.options?.[to];

    const generatorOpts: Partial<{ [K in AllGeneratorOptionKeys]: any }> = {
      ...generatorOptions,
      prettier: opts.prettier ?? true,
      plugins: [
        ...plugins,
        ...generatorOptions.plugins
      ],
```

**Review by samijaber — APPROVED** [review]
